### ISSUE TITLE
ASP.NET Core: config should be taken as argument

### ISSUE THREAD

**Issue by Tarmil** [dotnet-websharper/templates#19]
The ASP.NET Core templates currently create an `IConfiguration` by hand:

```fsharp
    member this.Configure(app: IApplicationBuilder, env: IHostingEnvironment) =
        // ...
        let config =
            ConfigurationBuilder()
                .SetBasePath(env.ContentRootPath)
                .AddJsonFile("appsettings.json")
                .Build()
        // use config...
```

This is both more verbose than necessary and less feature-complete, as it doesn't read `appsettings.<environment>.json`. Instead we can just receive it as argument:

```fsharp
    member this.Configure(app: IApplicationBuilder, env: IHostingEnvironment, config: IConfiguration) =
        // use config...
```

**Comment by Tarmil** [comment]
Given the latest changes to AspNetCore regarding DI, we should update Startup.fs even more substantially, basically making it quite similar to [the test project's](https://github.com/dotnet-websharper/aspnetcore/blob/bace29d77e8140faab62f99f707a3994e1976318/WebSharper.AspNetCore.Tests/Startup.fs).

Maybe we should also merge Startup.fs and Program.fs, I think their contents belong together.

**Comment by Tarmil** [comment]
Fixed in 308bc59da5490071c7eaab2ff2dcfe1efbd2ac3a
